Makefile

```diff
@@ -29,10 +29,52 @@ RESTY_LUAROCKS_VERSION ?= `grep RESTY_LUAROCKS_VERSION $(KONG_SOURCE_LOCATION)/.
 RESTY_OPENSSL_VERSION ?= `grep RESTY_OPENSSL_VERSION $(KONG_SOURCE_LOCATION)/.requirements | awk -F"=" '{print $$2}'`
 RESTY_PCRE_VERSION ?= `grep RESTY_PCRE_VERSION $(KONG_SOURCE_LOCATION)/.requirements | awk -F"=" '{print $$2}'`
 KONG_BUILD_TOOLS ?= '4.3.2'
-KONG_VERSION ?= `cat $(KONG_SOURCE_LOCATION)/kong-*.rockspec | grep tag | awk '{print $$3}' | sed 's/"//g'`
 OPENRESTY_PATCHES_BRANCH ?= master
 KONG_NGINX_MODULE_BRANCH ?= master
 
+PACKAGE_TYPE ?= deb
+REPOSITORY_NAME ?= kong-${PACKAGE_TYPE}
+REPOSITORY_OS_NAME ?= ${RESTY_IMAGE_BASE}
+KONG_PACKAGE_NAME ?= kong
+KONG_VERSION ?= `cat $(KONG_SOURCE_LOCATION)/kong-*.rockspec | grep -m1 tag | awk '{print $$3}' | sed 's/"//g'`
+
+BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+ISTAG = false
+ISPRERELEASE = false
+ifeq ($(BRANCH),HEAD)
+	ISTAG = true
+	BRANCH = $(shell git describe --tags --abbrev=0)
+	POSSIBLE_VERSION = $(shell git describe --tags --abbrev=0 | awk -F"-" '{print $$1}')
+	POSSIBLE_PRERELEASE_NAME = $(shell git describe --tags --abbrev=0 | awk -F"-" '{print $$2}')
+else
+	REPOSITORY_NAME = kong-${BRANCH}
+	REPOSITORY_OS_NAME = ${BRANCH}
+	KONG_PACKAGE_NAME = kong-${BRANCH}
+	KONG_VERSION = `date +%Y-%m-%d`
+endif
+ifneq ($(POSSIBLE_PRERELEASE_NAME),)
```

Comment:
It seems like POSSIBLE_PRERELEASE_NAME is only set on the `ifeq ($(BRANCH),HEAD)` branch above. Shouldn't we move this ifeq inside that branch?

Reply:
yes we should but a limitation of Make is we are not technically capable of doing this (correct me if I'm wrong)

Reply:
I can't find original sources (no explicit mention on the man pages), but I am finding non-canon sources like https://stackoverflow.com/a/51518105 which seem to indicate that nesting conditionals are a thing.

Reply:
adjusted in 67e9b4f